ui/src/app/components/cost-predictor.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+
+import colors, { colorWithWhiteness } from 'app/styles/colors';
+import { formatUsd } from 'app/utils/numbers';
+import { isUsingFreeTierBillingAccount } from 'app/utils/workspace-utils';
+
+import { FlexRow } from './flex';
+import { StartStopRuntimeButton } from './runtime-configuration-panel/start-stop-runtime-button';
+import { styles } from './runtime-configuration-panel/styles';
+import { RuntimeCostEstimator } from './runtime-cost-estimator';
+import { Spinner } from './spinners';
+
+const CostInfo = ({
+  creatorFreeCreditsRemaining,
+  runtimeChanged,
+  analysisConfig,
+  currentUser,
+  workspace,
+}) => {
+  const remainingCredits =
+    creatorFreeCreditsRemaining === null ? (
+      <Spinner size={10} />
+    ) : (
+      formatUsd(creatorFreeCreditsRemaining)
+    );
+
+  return (
+    <FlexRow data-test-id='cost-estimator'>
+      <div
+        style={{
+          padding: '.495rem .75rem',
+          ...(runtimeChanged
+            ? {
+                backgroundColor: colorWithWhiteness(colors.warning, 0.9),
+              }
+            : {}),
+        }}
+      >
+        <RuntimeCostEstimator {...{ analysisConfig }} />
+      </div>
+      {isUsingFreeTierBillingAccount(workspace) &&
+        currentUser === workspace.creator && (
+          <div style={styles.costsDrawnFrom}>
+            Costs will draw from your remaining {remainingCredits} of free
+            credits.
+          </div>
+        )}
+      {isUsingFreeTierBillingAccount(workspace) &&
+        currentUser !== workspace.creator && (
+          <div style={styles.costsDrawnFrom}>
+            Costs will draw from workspace creator's remaining{' '}
+            {remainingCredits} of free credits.
+          </div>
+        )}
+      {!isUsingFreeTierBillingAccount(workspace) && (
+        <div style={styles.costsDrawnFrom}>
+          Costs will be charged to billing account{' '}
+          {workspace.billingAccountName}.
+        </div>
+      )}
+    </FlexRow>
+  );
+};
+
+export const CostPredictor = ({
+  creatorFreeCreditsRemaining,
+  profile,
+  workspace,
+  analysisConfig,
+  status,
+  onPause,
```

Comment:
Is there a better way of doing getting on Pause and onResume to StartStopRuntimeButton rather than passing them through CostPredictor? Could we use context or a utility function or something?

Reply:
I don't think React is the issue here. If `CostPredictor` does more than just predict costs, maybe the component name should change instead? Or perhaps the component is doing too much?

Reply:
can you pass the StartStopRuntimeButton as a child component?  This would also simplify things because you can keep StartStopRuntimeButton runtime-specific (no changes) and pass in a "FakeCromwellStartButton" as a placeholder

Reply:
Hmmm, I like the idea of separating them, because they are quite distinct in their functionality, but since the two subcomponents are only ever used from within this component (not on their own), I will go the route of renaming.

I am willing to go the other way, if anyone would prefer that approach.

Reply:
EnvironmentInformedActionPanel. A little wordy, but I think it summarizes what the intent of the component is.